### ISSUE TITLE
TSM opens for only a few milliseconds 

### ISSUE THREAD

**Issue by gamdschiee** [sienori/Tab-Session-Manager#362]
## Short description
TSM window does not open, when I click on the small button top right corner of firefox.

## Expected result
Clicking the mouse fast so show, that the window only stays for a few milli-seconds open. I do not close it, it closes it by itself: 
![tsm_problem](https://user-images.githubusercontent.com/45832055/50547070-e87e4c00-0c32-11e9-827f-1b77690b3903.gif)

## Actual result
Normally there should be a small window, which shows my stored sessions. 

## Platform information
* Platform (OS): Windows 10
* Version of browser: Firefox Quantum 64.0 (64-Bit)
* Version of Tab Session Manager: 4.3.3

## Question
I easiest solution of my problem is probably just to reinstall TSM, but I am afraid I would lose all my stored sessions, so I am not really sure if they are even stored on my HDD. Can you help me out there please? Where can I find the stored sessions, when I have them stored at all and how can I maybe fix this problem without reinstalling?

**Comment by sienori** [comment]
Please enable debug mode and show me the logs when restarting Firefox and opening the popup.
https://github.com/sienori/Tab-Session-Manager/wiki/Q&A-How-to-use-debug-mode

**Comment by gamdschiee** [comment]
Thank for for fast responding! Sorry that I couldn't deliver the debug log faster: ![image](https://user-images.githubusercontent.com/45832055/50589583-8cf8cc00-0e87-11e9-90b8-a499635d2836.png)

**Comment by sienori** [comment]
The storage seems to be corrupted.
Please follow this page. 
https://github.com/sienori/Tab-Session-Manager/wiki/IndexedDB-Error#corrupted-file-exists-in-firefoxs-storage

**Comment by gamdschiee** [comment]
I tried that, but that didn't change anything unfortunately. Log from starting firefox and clicking the TSM button to answering this post here: 
![image](https://user-images.githubusercontent.com/45832055/50597839-0fde4e80-0ea9-11e9-860a-992682735f15.png)

**Comment by gamdschiee** [comment]
It works, but I haven't restored the Storage-folder yet. Which means I haven't got back my old stored session from early december and the rest of the last year.

**Comment by sienori** [comment]
Probably I think `1782160246ssensosi.sqlite` file is corrupted...

Please check: 
You said you edited the UUID of the sqlite file.
Is the UUID included in the sqlite file a UUID of the Tab Session Manager?
>6. Open `Browse Data` tab.  
  ![edit sqlite](https://raw.githubusercontent.com/wiki/sienori/Tab-Session-Manager/img/sqlite.png)
   The value of `origin` is the TSM directory name of the previous profile. (In this case: `moz-extension+++de50ed3a-7f77-40bc-af2e-c5953d30edbb`)  
   Change this value to the TSM directory name of the current profile. (In this case: `moz-extension+++ae28caf2-9857-4b99-bee9-9e2cfdd3b28c`)

https://github.com/sienori/Tab-Session-Manager/wiki/Q&A:-Sessions-data-storage-location;-How-to-migrate-sessions-data-to-another-profile

**Comment by gamdschiee** [comment]
Yes, the origin value is "moz-extension+++5539e5b9-347f-431b-a19e-182874bd7bc4". It is the same name as the folder has.

**Comment by sienori** [comment]
Unfortunately, I think it is difficult to restore lost session data.
Sorry for the inconvenience.

> By exporting a session you will create a .json file. Which means the program somehow gets the session out of this .sqlite databank file. So theoretically you could create several .json files out of this old sqlite file and import it back in.

Is there content in the json file? I think the contents are empty.

**Comment by gamdschiee** [comment]
Ye I can't export anything with the old .sqlite (corrupted) in the Default folder. I can only export it, when using this DB program. But when I try to import the .json files (on a new .sqlite of course) it does not import anything.

So is there really nothing to fix that corrupt .sqlite file? Not even looking for the name of the websites etc., because even that would be a huge help.

**Comment by irvinm** [comment]
You could try looking into your "history" for a list of websites you have visited for a basic list.  (It is a pain, but good if it is your last resort)

**Comment by RobertAb1** [comment]
@gamdschiee  _and others_

Vote on Bug 944918, Bug 1423917, Bug 1427928 _and other bugs_ on Bugzilla@Mozilla website to speed up solving this issue (with broken indexedDB) by Mozilla and removing this problem (_and other issues_) from TSM: #150

**Comment by gamdschiee** [comment]
@RobertAb1 So you thin my corrupt file is still savable when Mozilla helps or rather the chance that the problem occurs in the first hand will be way lower?

**Comment by RobertAb1** [comment]
@gamdschiee 
If your problem is related to broken indexedDB and the mentioned bugs will fixed well by Firefox developers than there is a chance that no new problems with this database will be in the future. But now Firefox developers have to fix bug discovered 5 years ago.
Broken indexedDB is also causing trouble with other extensions using it.

**Comment by RobertAb1** [comment]
https://github.com/sienori/Tab-Session-Manager/issues/358#issuecomment-451598302

**Comment by reyman** [comment]
Same error here, i try everything in the wiki ... and i continue to have the same error : `IndexedDB UnknownErr: ActorsParent.cpp:554 `

Waiting for fix in FF, is there a way to export some json correctly using `sqlitebrowser` to get my data back ? I try to read `object_data.json`, but data is binary...

**Comment by mechanikilla** [comment]
so this is still apparantly an ongoing issue??
i just followed all that, around and around, brain-splitting tutorial - fixes, and after every solution, the issue remained

its not just storage that's corrupted, its "1782160246ssensosi.sqlite" that's corrupted
